blockprotocol/blockprotocol
# Make the inclusion of mock data optional in MockBlockDock

## 🌟 What is the purpose of this PR?

MockBlockDock ships with sample entities and links to help jump-start block development and experimentation. While this is helpful in a lot of cases (especially when getting started and wanting to follow tutorials), it can prove annoying when trying to focus on block logic to do with entity creation. For example, if working with a block which creates links of `Date`s and `Event`s, the completely unrelated mock-data can get in the way.

As such this PR makes the data configurable through a simple toggle on the props of MBD.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203358502199087/1204080201525396/f) _(internal)_

## 🚫 Blocked by


## 🔍 What does this change?

- Adds an `includeProvidedMockData` prop on MBD
- Fills that in as `true` by default in block templates

## 📜 Does this require a change to the docs?

- Unsure if this is mentioned in tutorials or anywhere else.

## ⚠️ Known issues

- This implementation unfortunately splits the definition of the seed data in the datastore between multiple props (`initialData` and `includeProvidedMockData`). We will consider improving this in the near future to try and expose the mock data from the MBD package itself, and then spread that into the `initialData` in the templates. This will allow users to locally modify the mock data before putting it in the datastore, and _should_ help illustrate how to seed the datastore in the first place. This is made more complicated by necessary considerations about incidentally exposing the complexity of temporal versioning.

## 🐾 Next steps

- Try and unify the seed data into a single prop, as described above.

## 🛡 What tests cover this?

- `tsc` and `eslint` 
- Manual testing of MBD and the block templates

## ❓ How to test this?

1. Run `yarn workspace mock-block-dock dev` and see the seed data
2. Edit the `dev-app` of `mock-block-dock` and remove `includeProvidedMockData` and verify seed data is gone
3. Repeat the above for the block templates

## 📹 Demo

<img width="1368" alt="image" src="https://user-images.githubusercontent.com/25749103/224698479-04f6ea76-f37b-4681-a950-e4ea43a4ee1c.png">
<img width="1425" alt="image" src="https://user-images.githubusercontent.com/25749103/224698630-ab6c16c9-b1fe-446f-b9ca-3ccae6926edd.png">
